Remove hard-dependencies on packaging and pybind11,
which are only required for the build, not the actual installation

Author: Jan Reuter (JSC)

diff --color -Naur gmxapi-0.4.2.orig/setup.cfg gmxapi-0.4.2/setup.cfg
--- gmxapi-0.4.2.orig/setup.cfg	2023-06-01 19:44:02.000000000 +0200
+++ gmxapi-0.4.2/setup.cfg	2025-08-18 21:21:02.611944841 +0200
@@ -16,8 +16,6 @@
 packages = find:
 install_requires = 
 	mpi4py
-	packaging
-	pybind11
 	networkx>=2.0
 	numpy>1.7
 
